File: packages/react-native/ios/FounderRouteAnalyticsModule.m
#import <React/RCTBridgeModule.h>
@interface RCT_EXTERN_MODULE(FounderRouteAnalyticsModule, NSObject)
RCT_EXTERN_METHOD(configure:(NSString *)key endpoint:(NSString *)endpoint appId:(NSString *)appId properties:(NSArray *)properties traits:(NSArray *)traits verificationId:(NSString *)verificationId)
RCT_EXTERN_METHOD(setConsent:(BOOL)value)
RCT_EXTERN_METHOD(identify:(NSString *)identifier token:(NSString *)token traits:(NSDictionary *)traits)
RCT_EXTERN_METHOD(setAccount:(NSString *)identifier)
RCT_EXTERN_METHOD(track:(NSString *)name properties:(NSDictionary *)properties outcomeId:(NSString *)outcomeId)
RCT_EXTERN_METHOD(screen:(NSString *)name)
RCT_EXTERN_METHOD(reset)
RCT_EXTERN_METHOD(flush)
RCT_EXTERN_METHOD(getDiagnostics:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
@end

File: packages/react-native/ios/FounderRouteAnalyticsModule.swift
import Foundation
import React

@objc(FounderRouteAnalyticsModule)
final class FounderRouteAnalyticsModule: NSObject {
    private let analytics = FounderRouteAnalytics.shared
    @objc static func requiresMainQueueSetup() -> Bool { false }
    @objc func configure(_ key: String, endpoint: String, appId: String, properties: [String], traits: [String]) { analytics.configure(key: key, endpoint: endpoint, appId: appId, allowedProperties: properties, allowedTraits: traits) }
    @objc func setConsent(_ value: Bool) { analytics.setConsent(value) }
    @objc func identify(_ id: String, token: String?, traits: [String: Any]) { analytics.identify(id, token: token, traits: traits) }
    @objc func setAccount(_ id: String?) { analytics.setAccount(id) }
    @objc func track(_ name: String, properties: [String: Any], outcomeId: String?) { analytics.track(name, properties: properties, outcomeId: outcomeId) }
    @objc func screen(_ name: String) { analytics.screen(name) }
    @objc func reset() { analytics.reset() }
    @objc func flush() { analytics.flush() }
    @objc func getDiagnostics(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) { analytics.getDiagnostics { resolve($0) } }
}
